modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/MultiLevelROIGeometryProvider.java

```diff
@@ -31,22 +31,51 @@
 
     private final double inset;
 
+    /** The FootprintInsetPolicy if any for inset management */
     private final FootprintInsetPolicy insetPolicy;
 
+    /** 
+     * The optional granuleBounds to be used as a default value when there is 
+     * no need to do ROI lookup against specific features.
+     * Specify that when dealing with single granules store  
+     */
+    private final Geometry imposedGranuleBounds;
```

Comment:
Hmm... I don't understand if the name is misleading or if there is a real need for an override. If the features are null, this would be more of a defaultGranuleBounds?